src/EditorFeatures/VisualBasicTest/UseConditionalExpression/UseConditionalExpressionForReturnTests.vb

```diff
@@ -393,10 +393,27 @@ end class",
 "
 imports system.collections.generic
 
-class 
+class C
     iterator function M() as IEnumerable(of integer)
         Yield If(true, 0, 1)
     end function
+end class")
+        End Function
+
+        <WorkItem(36117, "https://github.com/dotnet/roslyn/issues/36117")>
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseConditionalExpression)>
+        Public Async Function TestMissingWhenCrossingPreprocessorDirective() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"
+class C
+    function M() as integer
+        dim check as boolean = true
+#if true
+        [||]if check
```

Comment:
Is this valid syntax for VB if statement?

Reply:
what part?  not having parens?  If so, yes this is valid VB :)

Reply:
I meant:

If check Then
    Return 3
End If

Reply:
ah.  the `then` is optional in VB.  Automatic code cleanup adds it.  but it's not necessary.

Reply:
But you would still need the End If?

Reply:
smacks-head.  sorry.  i get it now.  will add hte other test as well :)